### PR TITLE
Added go modules

### PR DESCRIPTION
**Pull request by wexel-nath**
* Added go.mod
* Update Dockerfile to use golang:1.11-stretch (was 1.10)
* Update Dockerfile to pull dependencies using go mod (was go get)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,9 @@
-FROM golang:1.10-stretch
+FROM golang:1.11-stretch
+ENV GO111MODULE=on
 WORKDIR /go/src/github.com/JumboInteractiveLimited/Gandalf
+COPY go.mod ./
```
Comment (Nekroze) [inline Dockerfile]: Should also include go.sum here otherwise the versions will change across builds.

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 language: go
 go:
   - "1.x"
-  - "1.9"
-  - "1.10"
```
Comment (Nekroze) [inline .travis.yml:5]: Could we keep 1.10 just to prove it still works without go modules?

**Review by Nekroze — APPROVED** [review]
Not sure if I am meant to have this kind of access, looking good though.

File: .travis.yml (hunk 2 above)
Comment (wexel-nath) [reply .travis.yml:5]: I'd like to, but getting this error in ci:
https://travis-ci.org/JumboInteractiveLimited/Gandalf/jobs/531592820
```
$ go get -t ./...
package github.com/jmartin82/mmock/definition: cannot find package "github.com/jmartin82/mmock/definition" in any of:
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/github.com/jmartin82/mmock/definition (from $GOROOT)
	/home/travis/gopath/src/github.com/jmartin82/mmock/definition (from $GOPATH)
```
Which is the reason why I'm updating to use modules, as it should be a quick fix.
Do you know of an easy way to get a specific version of a repo using go get?

Comment (wexel-nath) [reply .travis.yml:5]: I appreciate the feedback!

**Review by Nekroze — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (Nekroze) [reply .travis.yml:5]: ah I see, sadly not without modules, if we have to drop 1.10 thats fine, was just a hope :+1:

**Pull request merged**